api/Dockerfile

```diff
@@ -4,32 +4,46 @@ WORKDIR /app
 
 COPY . .  
 
-FROM node:18-alpine AS installer
+RUN npm update -g npm
+
+RUN npm config set registry https://registry.npmjs.com/
+
+RUN npm run preinstall
+
+RUN npm i --verbose --maxsockets 6
+
+RUN npm run build
+
+
+FROM node:18-alpine AS development
 
 WORKDIR /app
 
 COPY --from=builder /app/package*.json ./
+COPY --from=builder /app/ .
 
-COPY --from=builder /app/merge-extensions-deps.js ./
-
-COPY --from=builder /app/src/extensions ./src/extensions
+ENV NODE_ENV=development
 
-COPY --from=builder /app/patches ./patches
+RUN npm install
 
-RUN npm update -g npm
+EXPOSE 3000
 
-RUN npm config set registry https://registry.npmjs.com/
+CMD ["npm", "run", "start:debug"]
 
-RUN npm i --verbose --maxsockets 6
 
-FROM node:18-alpine AS runner
+FROM node:18-alpine AS production
 
 WORKDIR /app
 
-COPY --from=installer /app/ .
+COPY --from=builder /app/package*.json ./
+COPY --from=builder /app/ .
```

Comment:
Copying the entire /app directory may include unnecessary files, increasing image size. Copy only the necessary build artifacts or directories needed for the application to run.